metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableMetricModule.java

```diff
@@ -176,14 +134,13 @@ public BigtableMetricModule(
         maxWriteBatch = Math.max(MIN_MUTATION_BATCH_SIZE, maxWriteBatch);
         maxWriteBatch = Math.min(MAX_MUTATION_BATCH_SIZE, maxWriteBatch);
 
-        this.maxWriteBatchSize = maxWriteBatch;
-
-        this.mutateRpcTimeoutMs = mutateRpcTimeoutMs.orElse(DEFAULT_MUTATE_RPC_TIMEOUT_MS);
-        this.readRowsRpcTimeoutMs = readRowsRpcTimeoutMs.orElse(DEFAULT_READ_ROWS_RPC_TIMEOUT_MS);
-        this.shortRpcTimeoutMs = shortRpcTimeoutMs.orElse(DEFAULT_SHORT_RPC_TIMEOUT_MS);
-        this.maxScanTimeoutRetries = maxScanTimeoutRetries.orElse(DEFAULT_MAX_SCAN_TIMEOUT_RETRIES);
-        this.maxElapsedBackoffMs =
-         maxElapsedBackoffMs.orElse(DEFAULT_MAX_ELAPSED_BACKOFF_MILLIS);
+        this.metricsConnectionSettings = new MetricsConnectionSettingsModule(
```

Comment:
I'm not Dagger guru but this **new MetricsConnectionSettingsModule** from json properties looks suspicious for me. Don't you use **MetricsConnectionSettingsModule** to provide one?

Reply:
Yeah this probably is wonky/incorrect usage. I need to either provide it
via Dagger or pass it in - but not both. I will address and retest.

Reply:
Did you miss this one? It still using "new" where it should be using just @Inject

Reply:
Really? I just don’t see how that magic can work. I’ll try it tho as my understanding of Dagger 2 is still shaky at best.

Reply:
I’m trying but it’s NPE’ing in BigtableMetricModule because `this.metricsConnectionSettings` is null. I can’t see how `MetricsConnectionSettingsModule` can be constructed by Dagger 2. How can its constructor parameters, namely `maxWriteBatchSize mutateRpcTimeoutMs readRowsRpcTimeoutMs shortRpcTimeoutMs maxScanTimeoutRetries maxElapsedBackoffMs` get passed/provided? They’re only available from `com.spotify.heroic.metric.bigtable.BigtableMetricModule#BigtableMetricModule`, aren’t they?